### PR TITLE
Add the `unified_admin_domain` configuration option for the unified admin domain.

### PR DESCRIPTION
**Pull request by MatWrz**
### What this PR does

Currently, the install path assumes that the installation is done on `admin.shopify.com` if not in a spin environment. This prevent installation of the app against a local shopify server or mock server.

The PR provides a `unified_admin_domain` that is used when building the installation URL will allow for installs on non-spin and non-production environments.

### Checklist

Before submitting the PR, please consider if any of the following are needed:

- [x] Update `CHANGELOG.md` if the changes would impact users
- [ ] Update `README.md`, if appropriate.
- [ ] Update any relevant pages in `/docs`, if necessary
- [ ] For security fixes, the [Disclosure Policy](https://github.com/Shopify/shopify_app/blob/master/SECURITY.md#disclosure-policy) must be followed.

### CODE REVIEW

File: lib/shopify_app/controller_concerns/frame_ancestors.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ module FrameAncestors
       content_security_policy do |policy|
         policy.frame_ancestors(-> do
           domain_host = current_shopify_domain || "*.#{::ShopifyApp.configuration.myshopify_domain}"
-          "#{ShopifyAPI::Context.host_scheme}://#{domain_host} https://admin.shopify.com"
+          "#{ShopifyAPI::Context.host_scheme}://#{domain_host} https://admin.#{::ShopifyApp.configuration.unified_admin_domain}"
```
Comment (paulomarg) [inline lib/shopify_app/controller_concerns/frame_ancestors.rb:11]: Should we use the utils method here instead for consistency?
```suggestion
          "#{ShopifyAPI::Context.host_scheme}://#{domain_host} #{ShopifyApp::Utils::unified_admin_domain}"
```

Comment (paulomarg) [inline lib/shopify_app/controller_concerns/frame_ancestors.rb:11]: Also, should we add a test to `test/controllers/concerns/embedded_app_test.rb` to ensure that the ancestors are being set correctly?

**Review by paulomarg — APPROVED** [review]
Couple of nits, nothing blocking.

File: lib/shopify_app/controller_concerns/frame_ancestors.rb (hunk 1 above)
Comment (paulomarg) [reply lib/shopify_app/controller_concerns/frame_ancestors.rb:11]: Totally missed that it wasn't public 😅 - I am slightly concerned now that having both `unified_admin_path` and `unified_admin_domain` is going to lead to confusion. Maybe keep it this way for now and we think up a way of combining the two methods.

File: test/controllers/concerns/embedded_app_test.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 require "test_helper"
 require "action_view/testing/resolvers"
 
-class EmbeddedAppTest < ActionController::TestCase
+class EmbeddedAppTest < ActionDispatch::IntegrationTest
```
Comment (MatWrz) [inline test/controllers/concerns/embedded_app_test.rb:6]: This was needed for Rails to populate the `Content-Security-Policy` header
It is also encouraged by the Rails doc when testing controllers. (https://api.rubyonrails.org/classes/ActionController/TestCase.html)

>Rails discourages the use of functional tests in favor of integration tests (use [ActionDispatch::IntegrationTest](https://api.rubyonrails.org/classes/ActionDispatch/IntegrationTest.html)).

>New Rails applications no longer generate functional style controller tests and they should only be used for backward compatibility. Integration style controller tests perform actual requests, whereas functional style controller tests merely simulate a request. Besides, integration tests are as fast as functional tests and provide lot of helpers such as as, parsed_body for effective testing of controller actions including even [API](https://api.rubyonrails.org/classes/ActionController/API.html) endpoints.

File: lib/shopify_app/controller_concerns/frame_ancestors.rb (hunk 1 above)
Comment (MatWrz) [reply lib/shopify_app/controller_concerns/frame_ancestors.rb:11]: I've added some tests for the content security policy header.

**Pull request merged**